src/new_enums.py

```diff
@@ -0,0 +1,120 @@
+from enum import Enum, IntEnum, auto
+
+
+class PlayerState(IntEnum):         # useless
+    IDLE = 0
+    WALK = 1
+
+
+class ItemToUse(IntEnum):         # useless
+    REGULAR_TOOL = 0
+    SEED = 1
+
+
+class GameState(IntEnum):
+    MAIN_MENU = 0
+    LEVEL = 1
+    PAUSE = 2
+    SETTINGS = 3
+    SHOP = 4
+    EXIT = 5
+    GAME_OVER = 6
+    WIN = 7
+    CREDITS = 8
+
+
+class InventoryResource(Enum):
+    WOOD = auto()
+    APPLE = auto()
+    CORN = auto()
+    TOMATO = auto()
+    CORN_SEED = auto()
+    TOMATO_SEED = auto()
+
+
+ITEM_WORTHS = {
+    InventoryResource.WOOD: 8,
+    InventoryResource.APPLE: 4,
+    InventoryResource.CORN: 20,
+    InventoryResource.TOMATO: 40,
+    InventoryResource.CORN_SEED: 4,
+    InventoryResource.TOMATO_SEED: 5,
+}
+
+
+def get_item_worth(item: InventoryResource) -> int:
+    return ITEM_WORTHS[item]
+
+
+def is_inventory_seed(item: InventoryResource) -> bool:
+    return item in {InventoryResource.CORN_SEED, InventoryResource.TOMATO_SEED}
+
+
+class FarmingTool(Enum):
+    NONE = auto()
+    AXE = auto()
+    HOE = auto()
+    WATERING_CAN = auto()
+    CORN_SEED = auto()
+    TOMATO_SEED = auto()
+
+
+TOOL_TO_RESOURCE = {
+    FarmingTool.CORN_SEED: InventoryResource.CORN,
+    FarmingTool.TOMATO_SEED: InventoryResource.TOMATO
+}
+
+SWINGING_TOOLS = {FarmingTool.HOE, FarmingTool.AXE}
+
+
+def is_swinging_tool(tool: FarmingTool) -> bool:
+    return tool in SWINGING_TOOLS
+
+
+def is_farming_seed(tool: FarmingTool) -> bool:
+    return tool in {FarmingTool.CORN_SEED, FarmingTool.TOMATO_SEED}
+
+
+def convert_to_inventory_resource(tool: FarmingTool) -> InventoryResource:
+    return TOOL_TO_RESOURCE.get(tool, tool)
```

Comment:
And THIS is the bulk of the reason why I don't agree with the file's content.

Due to how savefiles are handled, the original enum implementation actually was tailored so that the instances could be directly converted into strings so savefiles could be easily debugged and read while testing.

Your implementation completely forgoes that, and while function names ARE more explicit, and well-typed, I feel your changes would pretty much require the save file system to be redone from scratch. Moreover, the usage of `enum.auto` does not actually let other programmers infer the different values of each member in an enum.

IntEnum members (and in consequence all members of its derived classes) can be used as regular ints, which is also why I chose this implementation. Instead of using up memory to store mappings when converting in-between enums, storing the sequences and then using the source enum to index consumes less memory.

Reply:
I'll delete it, then